### PR TITLE
Add multi-class AUC ROC.

### PR DESCRIPTION
**Pull request by deadly-panda**

### CODE REVIEW

**Review by eddiebergman — APPROVED** [review]
Code wise, this seems all okay to me but I didn't check the numerical values of the test! @mfeurer maybe you want to take a look when you can?

**Comment by eddiebergman** [comment]
Hi @deadly-panda,

I ran the automated tests and it all seems to pass except one test. Just a small change for the meta-data. You can view them by clicking on the **details** section on any of the failing tests.

You can run it locally with:
```python
pytest test/test_scripts/test_metadata_generation.py
```

**Comment by deadly-panda** [comment]
Hi @eddiebergman .. Thank you, if you can also check the discussion on the [issue](https://github.com/automl/auto-sklearn/issues/1482). 
Indeed, I had the same questions about the numerical values of the tests & some black issues in the sub-module.
I will check the failing test.

File: test/test_metric/test_metrics.py [hunk 1]
```diff
@@ -295,6 +295,63 @@ def test_threshold_scorer_multilabel(self):
         score = scorer(y_true, y_pred)
         self.assertAlmostEqual(score, -1.0)
 
+    def test_threshold_scorer_multi_class(self):
+        y_true = np.array([[0, 1], [1, 0], [1, 0], [0, 1]])
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py:299]: Hey, I'm afraid this is not testing what it's supposed to test. Right now, this tests for binary classification, which roc_auc can handle anyways. Could you please extend this to a multiclass problem?

File: test/test_metric/test_metrics.py [hunk 2]
```diff
@@ -295,6 +295,63 @@ def test_threshold_scorer_multilabel(self):
...
+            1,
+            0,
+            1,
+            {"average": "weighted"},
+        )
+        y_pred = np.array([[0.6, 0.4], [0.6, 0.4], [0.8, 0.2], [0.8, 0.2]])
+        score = scorer(y_true, y_pred)
+        self.assertAlmostEqual(score, 0.5)
+
+    def test_threshold_scorer_multilabel_multiclass(self):
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py:329]: To the best of my knowledge, there is no metric in scikit-learn that can handle this yet, so I would advise to remove this. In case there is any, could you please let us know which one it is?

File: test/test_metric/test_metrics.py [hunk 3]
```diff
@@ -344,6 +401,13 @@ def test_regression_metrics(self):
 
     def test_classification_metrics(self):
         for metric, scorer in autosklearn.metrics.CLASSIFICATION_METRICS.items():
+            if scorer in [
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py:404]: What's the reason to skip these?

File: test/test_metric/test_metrics.py [hunk 4]
```diff
@@ -404,6 +468,11 @@ def test_classification_binary(self):
                 "precision_samples",
                 "recall_samples",
                 "f1_samples",
+            ] or scorer in [
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py:471]: What's the reason to skip these?

File: test/test_metric/test_metrics.py [hunk 5]
```diff
@@ -451,6 +520,10 @@ def test_classification_multiclass(self):
             # Skip functions not applicable for multiclass classification.
             if metric in [
                 "roc_auc",
+                "roc_auc_ovo_macro_macro",
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py:523]: I'm surprised that you add these here. In my opinion, this should be working now because you add the strategies to deal with multiclass classification in roc_auc.

File: test/test_metric/test_metrics.py [hunk 6]
```diff
@@ -531,11 +604,14 @@ def test_classification_multiclass(self):
                 self.assertTrue(np.isfinite(score))
 
     def test_classification_multilabel(self):
-
         for metric, scorer in autosklearn.metrics.CLASSIFICATION_METRICS.items():
             # Skip functions not applicable for multi-label classification.
             if metric in [
                 "roc_auc",
+                "roc_auc_ovo_macro_macro",
```
Comment (mfeurer) [inline test/test_metric/test_metrics.py:611]: Why are these not applicable to multilabel classification?
